src/frontend/next/src/hooks/use-prefer-theme-colors.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import { Theme } from '@material-ui/core';
+import { darkTheme, lightTheme } from '../theme';
+
+export default function usePreferredThemeColors(): Theme | null {
+  const isDarkThemePreferred = useMediaQuery('(prefers-color-scheme: dark)');
+  const preferredTheme = useMemo(() => (isDarkThemePreferred ? darkTheme : lightTheme), [
+    isDarkThemePreferred,
+  ]);
+
+  // This portion of code only run on client
+  if (typeof window !== 'undefined') {
+    // First give priority to value stored in localStorage
+    const storedSelectedTheme = localStorage.getItem('theme');
+    if (storedSelectedTheme) {
```

Comment:
This if isn't necessary

Reply:
@humphd : My idea is whatever the value stored in localStorage should take priority. If it is the first time the user loaded the app, there is nothing in localStorage then we look at the `prefer-color-scheme` and then store that value to localStorage as initial value